python/ray/serve/_private/common.py

```diff
@@ -141,6 +180,244 @@ def update(
             message=message,
         )
 
+    def update_message(self, message: str):
+        return self._update(message=message)
+
+    def transition(
+        self,
+        transition: DeploymentStatusTransition,
+        message: str = "",
```

Comment:
`message` should actually be dependent on the target state, so it's weird that we pass it in from the outside (where we don't know what the end state would be)

Reply:
Let's call it `message_override` instead to make it more clear that this is only used to override the message when necessary

Reply:
The target state is actually only available on the "outside"; `DeploymentStatusInfo` doesn't have access (for example) to the current number of replicas, or what error happened, etc. I looked into renaming it to message_override, but realized that it's not overriding anything; this function doesn't actually make any decisions to set the message, the message is coming from "outside" 100% of the time.